### PR TITLE
Update to new particle system

### PR DESCRIPTION
**Pull request by MaxBorsch**
Requested by @Cervator

### CODE REVIEW

File: src/main/java/org/terasology/gooeysQuests/MagicBuildParticleClientSystem.java [hunk 1]
```diff
@@ -62,10 +62,11 @@ private void spawnMagicalBuildParticles(Region3i region) {
         EntityBuilder entityBuilder = entityManager.newBuilder(spawnDungeonParticlePrefab);
         LocationComponent locationComponent = entityBuilder.getComponent(LocationComponent.class);
         locationComponent.setWorldPosition(region.center());
-        BlockParticleEffectComponent particleEffect = entityBuilder.getComponent(BlockParticleEffectComponent.class);
+        PositionRangeGeneratorComponent particleEffect = entityBuilder.getComponent(PositionRangeGeneratorComponent.class);
         Vector3f size = new Vector3f(region.size().toVector3f());
         size.scale(0.5f);
-        particleEffect.spawnRange = size;
+        particleEffect.minPosition = size.mul(-1f);
+        particleEffect.minPosition = size;
```
Comment (Cervator) [inline src/main/java/org/terasology/gooeysQuests/MagicBuildParticleClientSystem.java]: Duplicate assignment? Should one of them have been `.maxPosition` ?

**Comment by Cervator** [comment]
Thanks :-)

@flo is this OK with you? Still a few other modules to fix before we can merge the engine PR first then modules

**Comment by flo** [comment]
@MaxBorsch thanks for the porting. I haven't tested it but I trust @MaxBorsch that it works out great. I hope we can merge his PR soon.

**Comment by Cervator** [comment]
Thanks @flo :-)

I seem to have discovered a multiplayer crash with the new particles + GooeysQuests (details in https://github.com/MovingBlocks/Terasology/issues/2728) so can't merge this for the play test just yet. I would love to include this in the testing but would need to figure out the crash bug, I expect it may trigger for any client-only player - but haven't validated that theory yet. That would make any kind of multiplayer session difficult :-)

**Comment by Cervator** [comment]
Not sure why this PR didn't auto-close when I merged. Oh well. Tested out and merged :-)

**Comment by flo** [comment]
You merged a modified version of the commit where you fixed a bug I think.

**Comment by Cervator** [comment]
Did that in the engine, yeah, but not here. the two commits look identical. They have different commit hashes though, somehow. Dunno why. Some pats of Git will forever remain a mystery to me :-)

**Pull request closed without merging**